### DISCUSSION TITLE
[Feature Request] Disable depth for embedded systems (gles)

### DISCUSSION THREAD

**Discussion by JeodC** [HarbourMasters/2ship2harkinian#556] [Ideas]
Libultraship allows building SoH and 2Ship with gles, however in 2Ship gles is unable to make use of depth animations such as the pulsing heart on HUD, the analog clock on HUD (final night works fine), the Pictobox, and possibly some other things. The deku bubble does work fine in this regard. For the aforementioned, they are outright invisible and the Pictobox photo is just a black box.

![image](https://github.com/HarbourMasters/2ship2harkinian/assets/47716344/43dd97aa-52e2-489f-89c4-182cc2b6ecd6)

**Comment by Malkierian** [comment]
I don't think this is something that would be incorporated in the main repo. It would be the responsibility of whatever dev started making the port for Android to implement and maintain on their fork.

**Reply by awe444** [reply]
To be fair, GLES has usage outside of Android, most notably ARM Linux distributions. Just like the OP I've also been able to build 2ship for an ARM Linux device with GLES support and I've seen the same minor graphical bugs. There's precedent for the project devs accepting PRs that add support for GLES on Linux (https://github.com/Kenix3/libultraship/pull/415) so hopefully someone will look into it eventually.

**Reply by JeodC** [reply]
Wouldn't it also be desired for console ports? Wiiu and switch use gles don't they?

**Comment by awe444** [comment]
The HUD issues are fixed by this PR:
https://github.com/Kenix3/libultraship/pull/611
The Picto Box seems like it will need a separate fix.
